pallets/process-validation/src/tests/create_process.rs

```diff
@@ -1,10 +1,142 @@
 use super::*;
+use crate::{Process, Version, ProcessStatus, Restriction::None, ProcessModel, VersionModel};
+use crate::Event::*;
+use frame_support::{assert_ok};
+use sp_std::prelude::*;
 
-use frame_support::assert_ok;
+// -- fixtures --
+#[allow(dead_code)]
+const PROCESS_ID1: [u8; 32] = [
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+];
+const PROCESS_ID2: [u8; 32] = [
+    1, 2, 3, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+];
 
 #[test]
-fn create_process_simple() {
+fn returns_error_if_origin_validation_fails_and_no_data_added() {
     new_test_ext().execute_with(|| {
-        assert_ok!(ProcessValidation::create_process(Origin::root()));
+        // update to check for error type
+        assert!(ProcessValidation::create_process(
+            Origin::none(),
+            PROCESS_ID1,
+            vec![{ None }]
+        ).is_err());
+
+        assert_eq!(<VersionModel<Test>>::get(PROCESS_ID1), 0);
+        assert_eq!(<ProcessModel<Test>>::get(PROCESS_ID1, 1), Process {
+            status: ProcessStatus::Disabled, 
+            restrictions: [].to_vec(),
+        });
+    });
+}
+
+#[test]
+fn if_no_version_found_it_should_return_default_and_insert_new_one() {
+    new_test_ext().execute_with(|| {
+        System::set_block_number(1);
+        assert_eq!(<VersionModel<Test>>::get(PROCESS_ID1), 0);
+        assert_ok!(ProcessValidation::create_process(
+            Origin::root(),
+            PROCESS_ID1,
+            vec![{ None }],
+        ));
+        assert_eq!(<VersionModel<Test>>::get(PROCESS_ID1), 1);
```

Comment:
You're only asserting against the `VersionModel` and not the `ProcessModel`. I'd probably make sure the actual process assertion happened correctly as well

Reply:
Updated, also checking `System:events()` The reason was that storage is mocked already and create_process has returned 'Ok()' so querying storage should give correct data, well at least this is what I think.